Modules/SubModule_ASL/xASL_wrp_RealignASL.m

```diff
@@ -94,8 +97,37 @@ function xASL_wrp_RealignASL(x,bSubtraction)
 xASL_delete(rpfile);
 
 % Run SPM
-if nFrames>2 && bSubtraction
+
+%####### Calculating the motion for every first TE of each PLD and
+% considering it the same for the other TEs from that PLD.
+%
+% if nFrames>2 && bSubtraction && length(x.EchoTime)>1  %Multi TE 
+%     uniqueTE=uniquetol(x.EchoTime); %gives the number of unique TEs
+%     NumTEs=numel(uniqueTE);
+%     minTE=min(uniqueTE); 
+%     positionMinTE=find(x.EchoTime == minTE); %positions that have the min TE
+%     ImInfo=spm_vol(InputPath);
+%     ImInfoFirstTEs=ImInfo(positionMinTE);
+%     
+%     if numel(unique(x.Q.Initial_PLD))==1 %multiTE + single PLD
+%         spm_realign(ImInfoFirstTEs,flags,true);
+%         MotionFirstTEs=load(rpfile);
+%         MotionAllTEs=repelem(MotionFirstTEs(:,:),NumTEs,1); %repeats each row NumTEs times

```

Comment:
@BeatrizPadrela Is this also possible with `repmat`? Is a bit more backward compatible.